CMakeLists.txt

```diff
@@ -1,89 +1,55 @@
-SET (OPENNS_PACKAGE_TARNAME "bag")
-SET (OPENNS_PACKAGE_URL "http://www.opennavsurf.org")
-SET (VERSION 1.6.3 OPENNS_PACKAGE_VERSION)
+# SWIG: use SWIG_MODULE_NAME property.
+if(POLICY CMP0086)
+    cmake_policy(SET CMP0086 NEW)
+endif()
 
-#-----------------------------------------------------------------------------
-# Option to Build Shared/Static libs, default is shared
-#-----------------------------------------------------------------------------
-OPTION (BUILD_SHARED_LIBS "Build Shared Libraries" ON)
-IF (BUILD_SHARED_LIBS)
-  SET (LIB_TYPE SHARED)
-ELSE (BUILD_SHARED_LIBS)
-  SET (LIB_TYPE STATIC)
-ENDIF (BUILD_SHARED_LIBS)
+# Disable addition of warning flags to the CMAKE_<LANG>_FLAGS by default for MSVC.
+if(POLICY CMP0092)
+    cmake_policy(SET CMP0092 NEW)
+endif()
 
-#-----------------------------------------------------------------------------
-# Option to Build against Shared/Static version of HDF5, default is shared
-#-----------------------------------------------------------------------------
-OPTION (BUILD_WITH_SHARED_HDF5 "Build with HDF5 as a shared library" ON)
+project(OPENNS LANGUAGES C CXX VERSION 2.0.0)
 
+# Options
+option(BAG_BUILD_SHARED_LIBS "Build Shared Libraries" ON)
+option(BAG_BUILD_SWIG "Build SWIG interface" OFF)
+option(BAG_BUILD_PYTHON "Build Python bindings" OFF)
+option(BAG_CI "The build is on the CI (Travis/Appveyor)." OFF)
 
-if (CMAKE_C_COMPILER_ID STREQUAL "GNU")
-    set (CMAKE_C_FLAGS "--std=gnu99 ${CMAKE_C_FLAGS}")
-    add_definitions ("-Wall")
-endif ()
+# Put the artifacts into a single directory for MSVC so running tests is easier.
+if(MSVC AND NOT BAG_CI)
+    if(NOT CMAKE_RUNTIME_OUTPUT_DIRECTORY)
+        set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/stage/${CMAKE_INSTALL_BINDIR})
+    endif()
 
-if (WIN32)
+    if(NOT CMAKE_LIBRARY_OUTPUT_DIRECTORY)
+        set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/stage/${CMAKE_INSTALL_LIBDIR})
+    endif()
 
-else()
-	# use, i.e. don't skip the full RPATH for the build tree
-	SET(CMAKE_SKIP_BUILD_RPATH  FALSE)
+    if(NOT CMAKE_ARCHIVE_OUTPUT_DIRECTORY)
+        set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/stage/${CMAKE_INSTALL_LIBDIR})
+    endif()
+endif()
 
-	# when building, don't use the install RPATH already
-	# (but later on when installing)
-	SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE) 
+if(CODE_COVERAGE)
+    include(CMakeModules/code-coverage.cmake)
+    # Exclude C wrapper for now since our tests only target C++ code currently
+    add_code_coverage_all_targets(EXCLUDE test/* api/bag.cpp api/bag.h)
+endif()
 
-	#SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+add_subdirectory(api)
+add_subdirectory(examples)
 
-	# add the automatically determined parts of the RPATH
-	# which point to directories outside the build tree to the install RPATH
-	SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+option(BAG_BUILD_TESTS "Build Tests" ON)
```

Comment:
Wouldn't it be easier to follow that we have all the options at the same spot? I usually have on a separated CMake file included. In this case, I am advocating for moving this option after BAG_BUILD_PYTHON.

Reply:
Agreed. We will declare all of the options in one place (perhaps in a separate file).

Reply:
Resolved.